packages/loader/container-loader/src/container.ts

```diff
@@ -841,8 +842,26 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
             this._resolvedUrl = resolvedUrl;
             await this.connectStorageService();
 
-            // This we can probably just pass the storage service to the blob manager - although ideally
-            // there just isn't a blob manager
+            // upload blobs here (NYI)
+
+            // post summary here
+            if (this.loader.services.detachedBlobStorage && this.loader.services.detachedBlobStorage.size > 0) {
+                if (this.cachedAttachSummary === undefined) {
+                    const appSummary: ISummaryTree = this.context.createSummary();
+                    const protocolSummary = this.captureProtocolSummary();
+                    this.cachedAttachSummary = combineAppAndProtocolSummary(appSummary, protocolSummary);
```

Comment:
I think we should always take summary in the detached container. Here we are taking the summary after we are firing the "attaching" event. We are taking the summary in both cases. So I think we can still take it in the detached container? and then just create empty file if need be, and then later call uploadSummary with the same detached summary?

Reply:
It depends on how we will implement blob upload. Given that we are not blocking user to make changes to container through async process, user actions may result in new blobs being added while empty file is created or some blobs are uploaded. One simple solution might be - keep uploading blobs until we have nothing to upload and only after that take a snapshot. Alternatively we can say that runtime need to wait until file creation is completed before proceeding with upload blobs that were initiated through this transition.

I think it would be good to look if we can reverse the flow toward push model, where runtime drives these processes, and loader only provides capabilities. I.e. loader calls IRuntime.createContainer(context: creationState), where creationState is some objects (interface) with some methods on it, and runtime drives the process till completion.